api/src/main/java/org/apache/brooklyn/api/catalog/CatalogConfig.java

```diff
@@ -34,5 +34,7 @@
      * a higher value appears higher in the list. the default is 1.
      * (negative values may be used to indicate advanced config which might not be shown unless requested.) */ 
     double priority() default 1;
-    
+
+     /** a pinned configuration means that it should be displayed, regardless of its priority */
+    boolean pinned() default false;
```

Comment:
It's not only `brooklyn-library`. Users of Brooklyn writing their own entities will have the same problem. The code will compile cleanly without warnings (even will be binary compatible for existing jars). The fields will just be invisible in UI.

Same applies (even stronger) to yaml. Making all fields invisible by default will break all existing blueprints.

For Java applying a `@CatalogConfig` annotation in 99% of the cases is done to show parameters in UI. In YAML adding top-level parameters is done for the same reason. That's why I think existing behaviour makes sense and we should keep it.